pkg/server/application_api/application_api_test: TestHotRanges2Response failed

pkg/server/application_api/application_api_test.TestHotRanges2Response [failed](https://tanzanite.cluster.engflow.com/invocations/default/047f7ccb-1b1a-4d77-8c15-754d0986846b?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL2FwcGxpY2F0aW9uX2FwaTphcHBsaWNhdGlvbl9hcGlfdGVzdA==) on master @ [a085c926b3ddeabb029d4ff5135d2165676ae7a2](https://github.com/cockroachdb/cockroach/commits/a085c926b3ddeabb029d4ff5135d2165676ae7a2):


```
=== RUN   TestHotRanges2Response
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestHotRanges2Response1477985108
    test_log_scope.go:76: use -show-logs to present logs inline
    storage_inspection_test.go:371: unexpected increase in qps between ranges; prev=0.00, current=1.34
    storage_inspection_test.go:371: unexpected increase in qps between ranges; prev=1.34, current=2.00
    storage_inspection_test.go:371: unexpected increase in qps between ranges; prev=2.00, current=3.25
    storage_inspection_test.go:375: -- test log scope end --
test logs left over in: outputs.zip/logTestHotRanges2Response1477985108
--- FAIL: TestHotRanges2Response (6.91s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestHotRanges2Response.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/server/application_api/storage_inspection_test.go
+++ b/pkg/server/application_api/storage_inspection_test.go
@@ -353,26 +353,31 @@ func TestHotRanges2Response(t *testing.T) {
 	if len(hotRangesResp.Ranges) == 0 {
 		t.Fatalf("didn't get hot range responses from any nodes")
 	}
-	lastQPS := math.MaxFloat64
+	cpuSupport := grunning.Supported
+	lastCPU := math.MaxFloat64
 	for _, r := range hotRangesResp.Ranges {
 		if r.RangeID == 0 {
 			t.Errorf("unexpected empty range id: %d", r.RangeID)
 		}
+
 		if r.QPS > 0 {
 			if r.ReadsPerSecond == 0 && r.WritesPerSecond == 0 && r.ReadBytesPerSecond == 0 && r.WriteBytesPerSecond == 0 {
 				t.Errorf("qps %.2f > 0, expected either reads=%.2f, writes=%.2f, readBytes=%.2f or writeBytes=%.2f to be non-zero",
 					r.QPS, r.ReadsPerSecond, r.WritesPerSecond, r.ReadBytesPerSecond, r.WriteBytesPerSecond)
 			}
-			// If the architecture doesn't support sampling CPU, it
-			// will also be zero.
-			if grunning.Supported && r.CPUTimePerSecond == 0 {
+
+			if cpuSupport && r.CPUTimePerSecond == 0 {
 				t.Errorf("qps %.2f > 0, expected cpu=%.2f to be non-zero", r.QPS, r.CPUTimePerSecond)
 			}
 		}
-		if r.QPS > lastQPS {
-			t.Errorf("unexpected increase in qps between ranges; prev=%.2f, current=%.2f", lastQPS, r.QPS)
+
+		// Ranges in response should be sorted by cpu.
+		if cpuSupport {
+			if r.CPUTimePerSecond > lastCPU {
+				t.Errorf("unexpected increase in cpu between ranges; prev=%.2f, current=%.2f", lastCPU, r.CPUTimePerSecond)
+			}
+			lastCPU = r.CPUTimePerSecond
 		}
-		lastQPS = r.QPS
 	}
 }
 

